p2-token-staking-fixed/src/lib.rs

```diff
@@ -406,10 +412,13 @@ impl Contract {
         if now >= self.farming_end {
             now = self.farming_end;
             // if at the end of farming we don't start a new round then we need to force a new round
+            // AUDIT_1: Should this be (self.farming_end - self.farming_start). Otherwise, it may
+            // be incorrectly adjust the round by 1, when it shouldn't be adjusted.
```

Comment:
no, because at this point `now = self.farming_end` and we basically want to check if we started a new round or not. Example, let's say our ticker is every 1s. Round len = 10. Start = 20, end = 35. Then we want to execute 2 rounds.